src/box/sql/vdbeInt.h

```diff
@@ -280,6 +286,15 @@ struct Vdbe {
 	 * Result set consists of two binding variables.
 	 */
 	uint32_t res_var_count;
+	/**
+	 * Number of variables in SQL_EXPR function. Can be 0, 1 or 'Many'.
+	 */
+	enum sql_expr_arg_count arg_count;
+	/**
+	 * Name of the only SQL_EXPR function variable. Invalid if
+	 * arg_count != ARG_COUNT_ONE.
+	 */
+	const char *arg_name;
```

Comment:
Why do you need all that? Can you just walk the `Vdbe.aOp` array when needed, find all those `OP_FetchByName`, and extract names from them? And fail if found more than one? IMO it would look simpler than these new highly specialized fields and how complicated it became to emit `OP_FetchByName`.

You should be able to do all that in `space_init_constraints()`. It validates that the functions exist, and you can also validate that the field constraints whose language is `SQL_EXPR` contain only one same name in all `OP_FetchByName` ops.

Reply:
Thank you for the suggestion! Done.